src/lib389/lib389/cli_ctl/dblib.py

```diff
@@ -48,6 +62,47 @@ def __init__(self, *args, **kwargs):
         self.__dict__ = self
 
 
+def get_bdb_impl_status():
+    # Check installed rpms
+    try:
+        from rpm import TransactionSet
+        ts = TransactionSet()
+        pkgs = { pkgname:(len ([ pkg for pkg in ts.dbMatch( 'name', \
+                 pkgname ) ]) > 0) for pkgname in TESTED_PACKAGES}
+        ts.closeDB()
+    except ImportError:
+        if os.path.isfile(RPM_TOOL):
+            pkgs = { pkgname:( subprocess.run([RPM_TOOL, '-q', pkgname], \
+                     stdin=subprocess.DEVNULL,stderr=subprocess.DEVNULL \
+                     ).returncode  == 0 ) for pkgname in TESTED_PACKAGES }
+        else:
+            pkgs = { pkg:None for pkg in TESTED_PACKAGES}
+
+    # Check libback-ldbm plugin content
+    p = Paths()
+    plugin = None
+    BDB_IMPL_STATUS
+    for pdir in PLUGIN_PATHS:
+        for ppath in glob.glob(f'{pdir}/libback-ldbm.*'):
+            plugin = ppath
+    if plugin and os.path.isfile(OBJDUMP_TOOL):
+        rc = subprocess.run(['objdump', '-R', plugin], text=True, capture_output=True)
+        symbs = { symb: symb in rc.stdout for symb in TESTED_SYMBOLS }
+    else:
+        symbs = { symb: None for symb in TESTED_SYMBOLS }
+
+    if pkgs[BDB_BUNDLED_RPM] is True:
+        return BDB_IMPL_STATUS.BUNDLED
+    if symbs[BDBRO_SYMBOL] is True:
+        return BDB_IMPL_STATUS.READ_ONLY
+    if pkgs[BDB_RPM] is not False and symbs[BDB_SYMBOL] is True:
+        return BDB_IMPL_STATUS.RPM
+    if symbs[BDB_SYMBOL] is False and pkgs[BDB_BUNDLED_RPM] is False:
```

Comment:
I think at this step pkgs[BDB_BUNDLED_RPM] is false so we may not test it.

Why if we have not the bdb symbols should we fallback to NONE. Having BDB_RPM is not enough to decide it is RPM ?

Reply:
No. FYI: the values have a 3 way logic:   True,False or None  (if we are not able to determine if 
 the rpm is installed or not  ==> not True does not mean False 
 
 No if ds389 is build to use Bundle RPM or Read-Obly bdb, even if someone somehow install libdb it will not be used because the backend is not linked with libdb  ...